stockApp/stockApp/Presentation/View/SearchAlertView.swift

```diff
@@ -0,0 +1,307 @@
+import UIKit
+import MaterialComponents.MaterialTextControls_FilledTextFields
+
+// MARK: - Class
+class SearchAlertView: UIView {
+
+    // MARK: - Properties
+    weak var delegate: SearchAlertViewDelegate?
+
+    private weak var callerView: UIView?
+    private lazy var viewModel: SearchViewModelProtocol = {
+        return SearchViewModel()
+    }()
+
+    private let backgroundView: UIView = {
+        let backgroundView = UIView()
+        backgroundView.backgroundColor = .black
+        backgroundView.alpha = Constants.startingAlpha
+        return backgroundView
+    }()
+
+    private let alertView: UIView = {
+        let alert = UIView()
+        alert.backgroundColor = .white
+        alert.layer.masksToBounds = true
+        alert.layer.cornerRadius = Constants.alertCornerRadius
+        alert.alpha = Constants.startingAlpha
+        return alert
+    }()
+
+    let titleUILabel: UILabel = {
+        let label = UILabel()
+        label.font = UIFont(name: Strings.heeboFont, size: UIValues.defaultFontSize)
+        label.font = label.font.withSize(Constants.titleFontSize)
+        label.text = Constants.alertTitle
+        label.textAlignment = .center
+        label.numberOfLines = Ints.labelNumberOfLines
+        label.textColor = UIColor.black
+        label.lineBreakMode = .byWordWrapping
+        label.layer.masksToBounds = false
+        return label
+    }()
+
+    let filterUITextField: UITextField = {
+        let textField = MDCFilledTextField(frame: UIScreen.main.bounds)
+        textField.label.font = UIFont(name: Strings.heeboFont, size: UIValues.defaultFontSize)
+        textField.label.font = textField.label.font.withSize(Constants.labelFontSize)
+        textField.label.text = Constants.filterLabelText
+        textField.label.lineBreakMode = .byWordWrapping
+        textField.label.numberOfLines = Ints.labelNumberOfLines
+        textField.label.textColor = UIColor.lightGray
+        textField.label.textAlignment = .center
+        textField.backgroundColor = Color.porcelainWhite
+        textField.setNormalLabelColor(UIColor.gray, for: .normal)
+        textField.autocapitalizationType = .none
+        textField.setFloatingLabelColor(Color.tundoraBlack, for: .normal)
+        return textField
+    }()
+
+    let filterPicker: UIPickerView = {
+        let pickerView = UIPickerView()
+        return pickerView
+    }()
+
+    let valueUITextField: MDCFilledTextField = {
```

Comment:
```suggestion
    let valueTextField: MDCFilledTextField = {
```

this is an example why you don't have to add Types in variableNames..

Reply:
done